src/graph/PermissionCheck.cpp

```diff
@@ -0,0 +1,139 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "graph/PermissionCheck.h"
+
+namespace nebula {
+namespace graph {
+
+/**
+ * Read space : kUse, kDescribeSpace
+ * Write space : kCreateSpace, kDropSpace, kCreateSnapshot, kDropSnapshot
+ *               kBalance, kAdmin, kConfig, kIngest, kDownload
+ * Read schema : kDescribeTag, kDescribeEdge,
+ *               kDescribeTagIndex, kDescribeEdgeIndex
+ * Write schema : kCreateTag, kAlterTag, kCreateEdge,
+ *                kAlterEdge, kDropTag, kDropEdge,
+ *                kCreateTagIndex, kCreateEdgeIndex, kDropTagIndex, kDropEdgeIndex,
+ * Read user :
+ * Write user : kCreateUser, kDropUser, kAlterUser,
+ * Write role : kGrant, kRevoke,
+ * Read data : kGo , kSet, kPipe, kMatch, kAssignment, kLookup,
+ *             kYield, kOrderBy, kFetchVertices, kFind
+ *             kFetchEdges, kFindPath, kLimit, KGroupBy, kReturn
+ * Write data: kBuildTagIndex, kBuildEdgeIndex,
+ *             kInsertVertex, kUpdateVertex, kInsertEdge,
+ *             kUpdateEdge, kDeleteVertex, kDeleteEdges
+ * Special operation : kShow, kChangePassword
+ */
+
+// static
+bool PermissionCheck::permissionCheck(session::Session *session, Sentence* sentence) {
+    auto kind = sentence->kind();
+    switch (kind) {
+        case Sentence::Kind::kUnknown : {
+            return false;
+        }
+        case Sentence::Kind::kUse :
+        case Sentence::Kind::kDescribeSpace : {
+            /**
+             * Use space and Describe space are special operations.
+             * Permission checking needs to be done in their executor.
```

Comment:
why? Miss the spaceId?

Reply:
> why? Miss the spaceId?

Yes, Usually we need to get the space id from the session when check permission. 
We can't get the space ID form session when `Use Space` or `Describe space `, so we need get the space id in their executor.